librz/include/rz_rop.h

```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: 2024 z3phyr <giridh1337@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#ifndef RZ_ROP_H
+#define RZ_ROP_H
+
+#include <rz_cmd.h>
+
+/**
+ * \file rz_rop.h
+ * \brief Return-Oriented Programming (ROP) related APIs and structures..
+ *
+ * This file contains definitions, structures, and function prototypes for handling ROP gadgets and constraints.
+ */
+
+/**
+ * \brief Information about a register.
+ */
+typedef struct rz_reg_info_t {
+	char *name;
+	bool is_mem_read; ///< Register involved in Memory read.
+	bool is_pc_write; ///< PC write flag.
+	bool is_var_read; ///< Register involved in Variable read.
+	bool is_var_write; ///< Register involved in Variable write.
+	bool is_mem_write; ///< Register involved in Memory write.
+	ut64 init_val;
+	ut64 new_val;
+} RzRegInfo;
```

Comment:
```suggestion
} RzRopRegInfo;
```

Otherwise it follows the naming in `rz_reg.h`.
Alos please document the last three fields.